qutebrowser/misc/crashsignal.py

```diff
@@ -430,6 +432,15 @@ def interrupt_really_forcefully(self, signum, _frame):
         print("WHY ARE YOU DOING THIS TO ME? :(")
         sys.exit(128 + signum)
 
+    def reload_config(self, _signum, _frame):
+        """Reload the config."""
+        log.signals.info("SIGHUP received, reloading config.")
+        config_commands = objreg.get('config-commands', from_command=True)
```

Comment:
I'm a little suspicious that calling a command via objreg then handling CommandError isn't the right thing to do here. I think the whole cmdutils layer was originally specifically meant as an interface for the UI and calling into it from other pieces of code feels a bit odd. (Although it's also pretty powerful so I suspect we'll review it for how extensions should call into it in the future.)

Anyhow, would you mind checking if duplicating some of the internals of `config_source()` is any worse? Eg `configfiles.read_config_py(standarddir.config_py())` and catch `ConfigFileErrors`?